stytch/consumer/api/sessions.py

```diff
@@ -0,0 +1,355 @@
+        if session_jwt is not None:
+            data["session_jwt"] = session_jwt
+
+        url = self.api_base.url_for("/v1/sessions/revoke", data)
+        res = self.sync_client.post(url, data)
+        return RevokeResponse.from_json(res.response.status_code, res.json)
+
+    async def revoke_async(
+        self,
+        session_id: Optional[str] = None,
+        session_token: Optional[str] = None,
+        session_jwt: Optional[str] = None,
+    ) -> RevokeResponse:
+        """Revoke a Session, immediately invalidating all of its session tokens. You can revoke a session in three ways: using its ID, or using one of its session tokens, or one of its JWTs. This endpoint requires exactly one of those to be included in the request. It will return an error if multiple are present.
+
+        Fields:
+          - session_id: The `session_id` to revoke.
+          - session_token: The session token to revoke.
+          - session_jwt: A JWT for the session to revoke.
+        """  # noqa
+        data: Dict[str, Any] = {}
+        if session_id is not None:
+            data["session_id"] = session_id
+        if session_token is not None:
+            data["session_token"] = session_token
+        if session_jwt is not None:
+            data["session_jwt"] = session_jwt
+
+        url = self.api_base.url_for("/v1/sessions/revoke", data)
+        res = await self.async_client.post(url, data)
+        return RevokeResponse.from_json(res.response.status, res.json)
+
+    def get_jwks(
+        self,
+        project_id: str,
+    ) -> GetJWKSResponse:
+        """Get the JSON Web Key Set (JWKS) for a Stytch Project.
+
+        Fields:
+          - project_id: The `project_id` to get the JWKS for.
+        """  # noqa
+        data: Dict[str, Any] = {
+            "project_id": project_id,
+        }
+
+        url = self.api_base.url_for("/v1/sessions/jwks/{project_id}", data)
+        res = self.sync_client.get(url, data)
+        return GetJWKSResponse.from_json(res.response.status_code, res.json)
+
+    async def get_jwks_async(
+        self,
+        project_id: str,
+    ) -> GetJWKSResponse:
+        """Get the JSON Web Key Set (JWKS) for a Stytch Project.
+
+        Fields:
+          - project_id: The `project_id` to get the JWKS for.
+        """  # noqa
+        data: Dict[str, Any] = {
+            "project_id": project_id,
+        }
+
+        url = self.api_base.url_for("/v1/sessions/jwks/{project_id}", data)
+        res = await self.async_client.get(url, data)
+        return GetJWKSResponse.from_json(res.response.status, res.json)
+
+    # MANUAL(authenticate_jwt)(SERVICE_METHOD)
+    # ADDIMPORT: from typing import Any, Dict, Optional
+    # ADDIMPORT: import jwt
+    # ADDIMPORT: import time
+    def authenticate_jwt(
+        self,
+        session_jwt: str,
+        max_token_age_seconds: Optional[int] = None,
+        session_custom_claims: Optional[Dict[str, Any]] = None,
+    ) -> Optional[Session]:
```

Comment:
I see we used to return an `AuthenticateResponse` for this function - why'd we change to just `Session`?

Reply:
It was a very... _weird_ `AuthenticateResponse` because it didn't actually include a `User` in the response. We were forced to make `User` optional in the old version _just_ because of this method, which isn't good. A remote authentication event will always return a User while a local one never will. More realistically, this local authentication is just giving us a `Session` object.

Reply:
Got it - I'll mark it in the spreadsheet as a change for the changelog